### PR TITLE
Login functionality

### LINKED ISSUE #19: Feature#6 Registered user messages
When the registration request is performed should show the next messages in an alert

- [x] Show error message if the user already exists
- [x] Show success message if successfully logged in

### LINKED ISSUE #20: Feature#7 Connection with user registration signature
Feature#7 Connection with user registration signature

The connection with the firebase should be performed in the following way

- [x] When registering, it must be verified that the user does not exist, if the user exists, deny the registration (optional)
- [x] When registering, create a record for user data in firestore and the necessary databases
- [x] If the user does not exist, send a verification email, once verified, send it to the login screen

Feature#5 Input control and submit function #18

The control of the inputs should be implemented with the next configuration

- [x]  The input change should be independent
- [x]  The value of the inputs can be accessed by the state

The submit function should

- [x]  Take the values of the inputs in an independent way
- [x]  Make the request of Firebase in async function and wait for success or failure

Feature#11 Connection with user log in signature 

The connection with the firebase should be performed in the following way

- [x] When login is successful should send to the home screen
- [x] When login is successful should make a registration of the user on redux
- [x] If the user does not exist, show an error message

Feature#9 Input control Login screen

The control of the inputs should be implemented with the next configuration

- [x]  The input change should be independent
- [x]  The value of the inputs can be accessed by the state

The submit function should

- [x]  Take the values of the inputs in an independent way
- [x]  Make the request of Firebase in async function and wait for success or failure

### LINKED ISSUE #25: Feature#10 Login messages
When the login request is performed should show the text messages in an alert

- [x] Show error message if the user already exists
- [x] Show success message if successfully logged in

### PR DESCRIPTION
**Pull request by DemisRincon**
Feature#7 Connection with user registration signature

The connection with the firebase should be performed in the following way
- [ ]  When registering, it must be verified that the user does not exist, if the user exists, deny the registration (optional)
- [ ]  When registering, create a record for user data in firestore and the necessary databases
- [ ]  If the user does not exist, send a verification email, once verified, send it to the login screen

Feature#5 Input control and submit function #18
The control of the inputs should be implemented with the next configuration
- [ ] The input change should be independent
- [ ]  The value of the inputs can be accessed by the state

The submit function should
- [ ] Take the values of the inputs in an independent way
- [ ]  Make the request of Firebase in async function and wait for success or failure

Feature#11 Connection with user log in signature
The connection with the firebase should be performed in the following way
- [ ] When login is successful should send to the home screen
- [ ]  When login is successful should make a registration of the user on redux
- [ ]  If the user does not exist, show an error message

Feature#9 Input control Login screen
The control of the inputs should be implemented with the next configuration
- [ ] The input change should be independent
- [ ] The value of the inputs can be accessed by the state

The submit function should
- [ ]  Take the values of the inputs in an independent way
- [ ] Make the request of Firebase in async function and wait for success or failure

### CODE REVIEW

File: src/containers/Login/actions/index.js [hunk 1]
```diff
@@ -1,17 +1,65 @@
...
   }
+  return;
+};
+
+export const signUp = data => async dispatch => {
+  const action = USER_SIGN_UP;
+
+  dispatch(request(action));
+  const response = await authFirebase.createUserWithMail(data);
+  console.log(auth().currentUser);
```
Comment (YilKanda) [inline src/containers/Login/actions/index.js]: I suggest removing this console.log()

File: src/containers/Login/layouts/index.js [hunk 2]
```diff
@@ -1,15 +1,16 @@
...
+import {useGoogleConfiguration, useRedirect} from '../../../library/hooks';
 
-import AuthContent from './authContent';
+import AuthContent from '../reduxContainer';
+import {Layout} from '../../../components';
 
-const Login = () => {
+const Login = props => {
+  useGoogleConfiguration();
+  //useRedirect(props);
```
Comment (YilKanda) [inline src/containers/Login/layouts/index.js]: If not necessary, you can delete this comment.

File: src/library/firebase/auth.firebase.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    .collection('complaints')
+    .doc('SbpNphb62pp4IEFhNsEo')
+    .get()
+    .then(resp => {
+      if (resp.exists) {
+        return resp.data()['listComplaints'];
+      }
+      return false;
+    });
+  console.log('RESPONSE=======>', fullData);
```
Comment (YilKanda) [inline src/library/firebase/auth.firebase.js]: I recommend removing this console.log()

File: src/library/firebase/index.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+// Firebase
+import authFirebase from "./auth.firebase";
+
+export {
+  // Firebase
```
Comment (YilKanda) [inline src/library/firebase/index.js]: If it doesn't take care, you can delete this comment.

File: src/navigators/screen.container.js [hunk 5]
```diff
@@ -6,47 +6,98 @@ import {
...
-    />
-  ),
+  headerLeft: () => {
+    if (HLeft) {
+      return (
+        <Icon
+          name="arrow-back-outline"
+          width={48}
+          height={48}
+          fill="#3366FF"
```
Comment (YilKanda) [inline src/navigators/screen.container.js]: In this part, the color can be placed as an imported variable.

File: src/navigators/screen.container.js [hunk 6]
```diff
@@ -6,47 +6,98 @@ import {
...
+      );
+    }
+  },
+  headerRight: () => {
+    if (HRight) {
+      return (
+        <Button
+          onPress={() => authFirebase.logoutFirebase(navigation.replace)}
+          title="🧑"
+          color={color ? color : '#B9D7F2'}
```
Comment (YilKanda) [inline src/navigators/screen.container.js]: In this part, the color can be placed as an imported variable.

**Review by YilKanda — COMMENTED** [review]
Just leave comments about some console.log () I found, and about some colors that can be imported from a Colors.js file, for example.

**Review by jkzasori — APPROVED** [review]

**Review by YilKanda — APPROVED** [review]
LGTM!

**Pull request merged**